returnn/util/file_cache.py

```diff
@@ -203,45 +200,64 @@ def cleanup(self, *, need_at_least_free_space_size: int = 0):
         for mtime, neg_size, fn in all_files:
             size = -neg_size
             delete_reason = None
-            if cur_time - mtime > self._cleanup_files_always_older_than_days * 60 * 60 * 24:
-                delete_reason = f"File is {(cur_time - mtime) / 60 / 60 / 24:.1f} days old"
-            else:
-                reached_more_recent_files = True
-            if not delete_reason and need_at_least_free_space_size > cur_expected_free:
-                # Still must delete some files.
-                if cur_time - mtime > cur_used_time_threshold:
-                    delete_reason = f"Still need more space, file is {(cur_time - mtime) / 60 / 60:.1f} hours old"
+
+            lock_dir, lock_file = self._get_lock_filename(fn)
```

Comment:
I wonder, we iterate here over all files, right? Does this include the lock files themselves? Then this logic here is wrong? We don't need another lock for the lock.

And what about info files, isn't this also wrong for them?

Reply:
We already exclude the info files, but we should also exclude lock files!

Reply:
I think our current solution is both easier and also less error-prone.